server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java

```diff
@@ -201,15 +203,17 @@ private void assertValidUpdate(List<MessageResult> messagesToBeUpdated, UpdateMe
             .collect(Guavate.toImmutableList());
         List<MailboxId> targetMailboxes = getTargetedMailboxes(previousMailboxes, updateMessagePatch);
 
-        boolean allMessagesWereDrafts = messagesToBeUpdated.stream()
+        boolean isDraft = messagesToBeUpdated.stream()
```

Comment:
Why did you changed this variable name ?

Reply:
Because that variable name was linked to the coupling I am trying to remove.

This variable name makes the assumption that this mail gets the $Draft keyword in all of its mailboxes to be a valid draft. It introduces coupling with the $Draft keyword merging function.

Furthermore, would you remember to update that piece of code the day we decide to change the $Draft merging function from intersection to union? My bet is no, and this might lead to failure when moving a JMAP $Draft message into Outbox.

To remove that coupling (and ensure all JMAP $Draft messages can be moved to Outbox), then we need to care about the message being a draft (which differs from all of its mailbox context having the $Draft keyword)

Reply:
Ok, I understand. What bothers me is the use of `messageS` variable name then, because it's not clear at all that we are looking for all the MailboxMessage instances of a given MessageId to decide if we can move it.
Can we try to rename variables to make the intent clearer ?

Reply:
Could you make a proposal? I already had a hard time finding these names and they are not strictly related to that PR imo.

Reply:
Done